src/main/java/ar/com/synergian/wagongit/Utils.java

```diff
@@ -11,15 +11,14 @@
 	private Utils() {
 	}
 
-	public static File createCheckoutDirectory(String path, boolean isPermanent) throws GitException {
+	public static File createCheckoutDirectory(String path) throws GitException {
 		File dir;
-		if (isPermanent) {
-			dir = new File(path);
-		} else {
+		if (path.endsWith(".git")) {
 			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(path));
+		} else {
+			dir = new File(path);
```

Comment:
This new feature of providing a directory, interferes somehow, in a "semantic" way, with the "safe checkout" flag (http://synergian.github.io/wagon-git/troubleshooting.html), which wouldn't have any sense any more. 

I don't like to have features that interfere one with each other. That's what makes me unhappy :)

I think it's much more elegant to provide a path, like you did. But I would like to remove the old flag so the features don't overlap. And also, as I spent quite some time writing the documentation, I'd like to have it updated to these changes.

I'm not asking you to make those changes, I can. Sadly, I don't have the time right now, so they would have to wait.